DisGeNET.pm

```diff
@@ -0,0 +1,202 @@
+
+ gunzip all_gene_disease_pmid_associations.tsv.gz
+
+ awk '($1 ~ /^snpId/ || $2 ~ /NA/) {next} {print $0 | "sort -k2,2 -k3,3n"}' 
+ all_gene_disease_pmid_associations.tsv > all_variant_disease_pmid_associations_sorted.tsv
+
+ bgzip all_variant_disease_pmid_associations_sorted.tsv
+ tabix -s 2 -b 3 -e 3 all_variant_disease_pmid_associations_sorted.tsv.gz
+
+ The plugin can then be run as default (Option 1):
+ ./vep -i variations.vcf --plugin DisGeNET,/path/to/disgenet_file.tsv.gz
+
+ or with an option to include disease/phenotype data (Option 2): 
+ ./vep -i variations.vcf --plugin DisGeNET,/path/to/disgenet_file.tsv.gz,1
+
+ Of notice: this plugin only matches the chromosome and the position in the
+  chromosome, the alleles are not taken into account to append the DisGENET data.
+  The rsid is provided in the output in order to help to filter the relevant data.
+
+
+=cut
+package DisGeNET;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub new {
+  my $class = shift;
+
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+
+  $self->get_user_params();
+
+  if(defined($self->params->[1])) {
+    $self->{disease} = $self->params->[1];
+  }
+
+  return $self;
+}
+
+sub feature_types {
+  return ['Feature','Intergenic'];
+}
+
+sub get_header_info {
+  my $self = shift;
+
+  my %header;
+
+  $header{'DisGeNET'} = 'Publications reporting the Variant-Disease association. Output includes three values pmid:rsid:score. pmid - is the pmid of the publication reporting the Variant-Disease association; rsid - dbSNP variant Identifier; score - DisGENET score for the Variant-Disease association';
+
+  if($self->{disease}) {
+    $header{'DisGeNET_disease'} = 'Name of the disease reporting the Variant-pmid association';
+  }
+
+  return \%header;
+}
+
+sub run {
+  my ($self, $tva) = @_;
+
+  my $vf = $tva->variation_feature;
+
+  my $chr = $vf->{chr};
+
+  my $end = $vf->{end};
+  my $start = $vf->{start};
+  ($start, $end) = ($end, $start) if $start > $end;
+
+  my @data = @{$self->get_data($chr, $start, $end)};
+
+  return {} unless(@data);
+
+  my %hash;
+  my @result;
+  my %unique_values;
+  my @diseases;
+  my %unique_diseases;
+
+  my $n_results = 1;
+
+  foreach my $data_value (@data) {
+    my $pmid = $data_value->{pmid};
+    my $rsid = $data_value->{rsid};
+    my $score = $data_value->{score};
+
+    # Some publications are duplicated - same publications from different sources are in different rows
+    # Check if pmid and rsid are not returned more than once
+    if(!$unique_values{$pmid.':'.$rsid}++) {
+      my $result_string = $pmid . ':' . $rsid . ":" . $score;
+      push @result, $result_string;
+    }
+
+    if($self->{disease}) {
+      my $disease_name = $data_value->{diseaseName};
+      if(!$unique_diseases{$disease_name}++) {
+        push @diseases, $disease_name;
```

Comment:
If there is a comma in the disease name, the results look odd eg. rs587777299

Reply:
I replaced the comma by '&'. Now all output formats have the same.

Reply:
The '&' is used to separate different disease names in VCF output, so these now look odd-

20301347&22205389|0.70&0.70|Myasthenic_Syndromes&_Congenital
&MYASTHENIC_SYNDROME&_CONGENITAL&_8

An underscore may be better.